test/cache_invalidation_test.rb

```diff
@@ -211,4 +211,109 @@ def test_cache_not_expired_until_after_transaction
     assert_match(/^COMMIT\b/, sql_queries[2])
     assert_match(/^cache_write.active_support /, log.last.last)
   end
+
+  def test_expire_cache_with_primary_key_only
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, read_entity)
+    assert(@record.expire_cache)
+    assert_equal(IdentityCache::DELETED, read_entity)
+  end
+
+  def test_expire_cache_with_primary_key_only_on_failure
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, read_entity)
+
+    IdentityCache.cache_backend = CacheConnection.unconnected_cache_backend
+
+    refute(@record.expire_cache)
```

Comment:
I think we want to use the normal cache connection after the expiring the cache for `read_entity` on the following line.  Otherwise, `refute_equal(IdentityCache::DELETED, read_entity)` could just pass because `read_entity` is returning `nil` when the cache isn't present.

There are a few places where `IdentityCache.cache_backend = CacheConnection.unconnected_cache_backend` is done that make the same mistake, so that may as well be done in a block method that resets the connection.

```ruby
  def with_unconnected_cache_backend
    IdentityCache.cache_backend = CacheConnection.unconnected_cache_backend
    yield
    IdentityCache.cache_backend = backend
  end
```

```suggestion
    with_unconnected_cache_backend do
      refute(@record.expire_cache)
    end
```

Reply:
@dylanahsmith to be honest in this particular examples it doesn't make much sense the `unconnected_backend`. It's fine to use it once to see if true/false is returned, but then it makes more sense to stub the server to mimic a single fluke, if we switch it back and forth it doesn't make sense, as we now that the `expire_cache` hit a completely different backend anyway, so there's no reason to believe that the initial values would be different. 

I'll look a little bit more into this, and fallback to stubs if I don't find anything suitable.

Reply:
I'm not saying this is a replacement for stubbing.  Just that where we do use the unconnected_cache_backend, the assertions that follow should make sense.  For the code I was commenting on, it wasn't clear what those `refute_equal` statements were testing.

Reply:
This makes sense! I've also adjusted rest of the tests to assert for correct value instead of refuting `DELETED` key. Sorry @dylanahsmith for a lot of back and forth, I'm still new to minitest approach 🙇‍♂️